glfw/src/Window.zig

```diff
@@ -205,66 +210,119 @@ pub const Hint = enum(c_int) {
-                    }
-                },
-                else => @compileError("expected pointer to array, got " ++ @typeName(pointed_type)),
+/// Window hints
+pub const Hints = struct {
+    resizable: bool = true,
+    visible: bool = true,
+    decorated: bool = true,
+    focused: bool = true,
+    auto_iconify: bool = true,
+    floating: bool = false,
+    maximized: bool = false,
+    center_cursor: bool = true,
+    transparent_framebuffer: bool = false,
+    focus_on_show: bool = true,
+    scale_to_monitor: bool = false,
+    
+    red_bits: c_int = 8,
+    green_bits: c_int = 8,
+    blue_bits: c_int = 8,
+    alpha_bits: c_int = 8,
+    depth_bits: c_int = 24,
+    stencil_bits: c_int = 8,
+    accum_red_bits: c_int = 0,
+    accum_green_bits: c_int = 0,
+    accum_blue_bits: c_int = 0,
+    accum_alpha_bits: c_int = 0,
+    aux_buffers: c_int = 0,
+    samples: c_int = 0,
+    refresh_rate: c_int = consts.dont_care,
+    
+    stereo: bool = false,
+    srgb_capable: bool = false,
+    doublebuffer: bool = true,
+    
+    client_api: ClientApi = .opengl_api,
+    context_creation_api: ContextCreationApi = .native_context_api,
+    
+    context_version_major: c_int = 1,
+    context_version_minor: c_int = 0,
+    
+    context_robustness: ContextRobustness = .no_robustness,
+    context_release_behavior: ContextReleaseBehavior = .any_release_behavior,
+    
+    opengl_forward_compat: bool = false,
+    opengl_debug_context: bool = false,
+    
+    opengl_profile: OpenGlProfile = .opengl_any_profile,
+    
+    cocoa_retina_framebuffer: bool = true,
+    
+    cocoa_frame_name: [:0]const u8 = "",
+    
+    cocoa_graphics_switching: bool = false,
+    
+    x11_class_name: [:0]const u8 = "",
+    x11_instance_name: [:0]const u8 = "",
+    
+    pub const ClientApi = enum(c_int) {
+        opengl_api = consts.opengl_api,
+        opengl_es_api = consts.opengl_es_api,
+        no_api = consts.no_api,
+    };
+    
+    pub const ContextCreationApi = enum(c_int) {
+        native_context_api = consts.native_context_api,
+        egl_context_api = consts.egl_context_api,
+        osmesa_context_api = consts.osmesa_context_api,
+    };
+    
+    pub const ContextRobustness = enum(c_int) {
+        no_robustness = consts.no_robustness,
+        no_reset_notification = consts.no_reset_notification,
+        lose_context_on_reset = consts.lose_context_on_reset,
+    };
+    
+    pub const ContextReleaseBehavior = enum(c_int) {
+        any_release_behavior = consts.any_release_behavior,
+        release_behavior_flush = consts.release_behavior_flush,
+        release_behavior_none = consts.release_behavior_none,
+    };
+    
+    pub const OpenGlProfile = enum(c_int) {
+        opengl_any_profile = consts.opengl_any_profile,
+        opengl_compat_profile = consts.opengl_compat_profile,
+        opengl_core_profile = consts.opengl_core_profile,
+    };
```

Comment:
Personally I'd remove these constants from consts and inline them here. They don't need to be publicly accessible any more, so there's probably no reason for them to be in consts.

Reply:
@silversquirl I was thinking the same thing, but was unsure whether to do so being that something like this wasn't done in #56 already.

Reply:
@silversquirl I went ahead with the change; the only constant in consts.zig now is `dont_care`.